File: app_gestion_activos/lib/models/EstadoDispositivo.dart
class EstadoDispositivo {
  final int idEstadoDispositivo;
  final String nombreEstadoDispositivo;
  final String? descripcionEstadoDispositivo;
  final int? estadoDispositivoCreadoPorId;

  EstadoDispositivo({
    required this.idEstadoDispositivo,
    required this.nombreEstadoDispositivo,
    this.descripcionEstadoDispositivo,
    this.estadoDispositivoCreadoPorId,
  });

  factory EstadoDispositivo.fromJson(Map<String, dynamic> json) {
    return EstadoDispositivo(
      idEstadoDispositivo: json['idEstadoDispositivo'] ?? 0,
      nombreEstadoDispositivo: json['nombreEstadoDispositivo'] ?? '',
      descripcionEstadoDispositivo: json['descripcionEstadoDispositivo'],
      estadoDispositivoCreadoPorId: json['estadoDispositivoCreadoPorId'],
    );
  }

  Map<String, dynamic> toJson() => {
    'idEstadoDispositivo': idEstadoDispositivo,
    'nombreEstadoDispositivo': nombreEstadoDispositivo,
    'descripcionEstadoDispositivo': descripcionEstadoDispositivo,
    'estadoDispositivoCreadoPorId': estadoDispositivoCreadoPorId,
  };
}

File: services/inventory-service/models/estadoDispositivo.js
const { DataTypes } = require("sequelize");

module.exports = (sequelize, DataTypes) => {
  const EstadoDispositivo = sequelize.define(
    "EstadoDispositivo",
    {
      idEstadoDispositivo: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true,
        allowNull: false,
      },
      nombreEstadoDispositivo: {
        type: DataTypes.STRING,
        allowNull: false,
      },
      descripcionEstadoDispositivo: {
        type: DataTypes.STRING,
        allowNull: true,
      },
      estadoDispositivoCreadoPorId: {
        type: DataTypes.INTEGER,
        allowNull: true,
      }
    },
    {
      tableName: "estadosdispositivos",
      timestamps: true, // Para manejar createdAt y updatedAt automáticamente
    }
  );
  return EstadoDispositivo;
};
